crates/sui-types/src/error.rs

```diff
@@ -536,9 +548,8 @@ impl From<&str> for SuiError {
 
 impl SuiError {
     pub fn indicates_epoch_change(&self) -> bool {
-        let err_str = self.to_string();
-        err_str.contains(VALIDATOR_HALTED_ERROR_MSG)
-            || err_str.contains(MISSING_COMMITTEE_ERROR_MSG)
+        matches!(self, SuiError::ValidatorHaltedAtEpochEnd)
+            || matches!(self, SuiError::MissingCommitteeAtEpoch(_))
```

Comment:
I don't remember exactly ,but these two types of errors can be in a vector formed SuiError or purely in a string, e.g.  https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/authority_aggregator.rs#L1723 and https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/authority_aggregator.rs#L1880

Reply:
some marginal improvement https://github.com/MystenLabs/sui/pull/5723